gui/wxpython/startup/utils.py

```diff
@@ -55,6 +58,54 @@ def get_possible_database_path():
     return path
 
 
+def create_possible_database_path():
+    """Create the directory to what is possibly a GRASS Database.
+
+    Create directory named grassdata in the usual locations.   
+    
+    Returns the new path as a string.  
+    """
+    home = os.path.expanduser('~') 
+    tmp = os.path.join(tempfile.gettempdir(),
+                               "grassdata_{}".format(getpass.getuser()))           
+    
+    # Create independent "grassdata" in the home Linux dir
+    if sys.platform.startswith('linux'):
+        try:
+            path = os.path.join(home, "grassdata")
+            os.mkdir(path)
+        except OSError: 
```

Comment:
The `path` is already set at this point even when exception was raised, so your line `if path is None:` will have path which is not None and thus the functions will just return this path.